src/timer.ts

```diff
@@ -0,0 +1,108 @@
+import { Entity } from '@dcl/sdk/ecs'
+import { getTriggerEvents } from './events'
+import { TriggerType } from './enums'
+
+interface DelayAction {
+  action: string
+  timeout: number
+  callback: Function
+}
+
+interface IntervalAction {
+  action: string
+  timeout: number
+  interval: number
+  callback: Function
+}
+
+const queueDelay = new Map<Entity, Array<DelayAction>>()
+const queueInterval = new Map<Entity, Array<IntervalAction>>()
+
+export function createTimerSystem() {
+  return function timerSystem(dt: number) {
+    intervalSystem(dt)
+    delaySystem(dt)
+  }
+
+  function intervalSystem(dt: number) {
+    for (const [entity, actions] of queueInterval.entries()) {
+      const triggerEvents = getTriggerEvents(entity)
+
+      for (const action of actions) {
+        if (action.timeout === action.interval) {
+          action.callback()
+        }
+
+        action.timeout -= dt
+
+        if (action.timeout <= 0) {
+          action.timeout = action.interval
+          triggerEvents.emit(TriggerType.ON_LOOP)
+        }
+      }
+    }
+  }
+
+  function delaySystem(dt: number) {
+    for (const [entity, actions] of queueDelay.entries()) {
+      const triggerEvents = getTriggerEvents(entity)
+      const completedActions = []
+      let idx = 0
+
+      // Iterate entity actions and verify is the timeout is reached to execute the callback
+      for (const action of actions) {
+        action.timeout -= dt
+
+        if (action.timeout <= 0) {
+          action.callback()
+          triggerEvents.emit(TriggerType.ON_DELAY)
+          completedActions.push(idx)
+        }
+        idx++
+      }
+
+      // Remove completed actions from the queue
+      for (const action of completedActions) {
+        actions.splice(action, 1)
+      }
+    }
+  }
+}
+
+export function setTimeout(
```

Comment:
I think we should avoid using the same name as the native javascript `setTimeout` and `setInterval` methods, even though they are not available in the SDK (they are not available in the typing, but in runtime they are, because the runtime is the browser's javascript environment).

What do you think about `startTimeout` and `startInterval` instead?